src/sinks/util/service.rs

```diff
@@ -360,7 +361,7 @@ impl TowerRequestSettings {
 
         // Build services
         let open = OpenGauge::new();
-        let max_concurrency = services.len() * AdaptiveConcurrencySettings::max_concurrency();
+        let max_concurrency = services.len() * default_request_builder_concurrency_limit();
```

Comment:
I'm not 100% sure if this is the correct multiplier to use, but it _feels_ right. I'm hoping someone else will have thoughts.

Reply:
At a higher level, I'm questioning whether or not we need yet another buffer here or if the buffer layer can be removed altogether.

Reply:
Thinking about it more, I don't think its necessary and it adds an additional place for data to buffer. I'm in favor of removing it if there are no objections.